src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -101,7 +102,6 @@ public GoogleLoginState(
 
     this.isLoggedIn = false;
     this.email = "";
-    connected = true; // assume we're connected until checkCredentials is called
 
     listeners = Lists.newLinkedList();
```

Comment:
In fact, nothing is thread-safe. For this reason, we are using locks at the client-side (CT4E), so we are okay. Actually, CT4E don't add a listener at all (until now), so we are really okay for now. I haven't thought much about listeners for this reason.

I'll have a chance to make this class thread-safe later (hopefully), and when I do, I'll have to think about the listener list seriously.